python/beeai_framework/serializer/serializable.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2025 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import Any, TypeVar
+
+from pydantic import BaseModel
+
+T = TypeVar("T", bound="Serializable")
+
+
+class Serializable(ABC):
+    """Base class for serializable objects."""
+
+    @classmethod
+    async def from_snapshot(cls: type[T], snapshot: dict[str, Any]) -> T:
+        """Create instance from snapshot."""
+        if issubclass(cls, BaseModel):
+            return cls(**dict(snapshot))
+        else:
+            instance = cls()
+            await instance.load_snapshot(snapshot)
+            return instance
```

Comment:
This will not work. Some classes can't be initiated without parameters.

Reply:
following our discussion... `Serializable` has been simplified to only include an abstract `clone` method.

https://github.com/i-am-bee/beeai-framework/pull/705/commits/38e91df948e51d41c5e353bea7470d3bf8e1c32c